src/middlewared/middlewared/alert/service/telegram.py

```diff
@@ -0,0 +1,33 @@
+import json
+import requests
+import re
+
+from middlewared.alert.base import ThreadedAlertService
+from middlewared.schema import Dict, Str
+
+
+class TelegramAlertService(ThreadedAlertService):
+    title = "Telegram"
+
+    schema = Dict(
+        "telegram_attributes",
+        Str("bot_token", required=True, empty=False),
+        Str("chat_ids", required=True, empty=False),
+        strict=True,
+    )
+
+    def send_sync(self, alerts, gone_alerts, new_alerts):
+        token = self.attributes["bot_token"]
+        chat_ids = [c_id.strip() for c_id in re.findall(r"-?[\d']+", self.attributes["chat_ids"])]
```

Comment:
Looks great now, thank you! But why is `'` included? Also you won't need `.strip()` because your regex does not match any whitespace characters

Reply:
Yeah, you're right! I fix these two things. Thanks for the tips.

Reply:
Sorry, can you please use `List("chat_ids", required=True, empty=False, items=[Int("chat_id")]),` instead and remove all the parsing? We'll do it in the UI instead.